File: frontend-hospital-management/src/services/prescriptionService.js
import api from './api'

const prescriptionService = {
  getAllPrescriptions() {
    return api.get('/prescriptions')
  },

  getPrescriptionById(id) {
    return api.get(`/prescriptions/${id}`)
  },

  createPrescription(prescription) {
    return api.post('/prescriptions', prescription)
  },

  updatePrescription(id, prescription) {
    return api.put(`/prescriptions/${id}`, prescription)
  },

  deletePrescription(id) {
    return api.delete(`/prescriptions/${id}`)
  }
}

export default prescriptionService

File: src/main/java/com/enesgunumdogdu/hospital_management/service/PrescriptionService.java
package com.enesgunumdogdu.hospital_management.service;

import com.enesgunumdogdu.hospital_management.domain.dto.PrescriptionDto;

import java.util.List;

public interface PrescriptionService {
    
    List<PrescriptionDto> getAllPrescriptions();
    
    PrescriptionDto getPrescriptionById(Long id);
    
    PrescriptionDto createPrescription(PrescriptionDto prescriptionDto);
    
    PrescriptionDto updatePrescription(Long id, PrescriptionDto prescriptionDto);
    
    void deletePrescription(Long id);
}

